core/src/main/java/com/linecorp/armeria/common/DefaultConcurrentAttributes.java

```diff
@@ -108,7 +109,9 @@ private <T> T attr(AttributeKey<T> key, boolean ownAttr) {
             return null;
         }
 
-        synchronized (head) {
+        final ReentrantLock reentrantLock = new ReentrantLock();
```

Comment:
This doesn't mean anything.
`synchronized (head)` was used to prohibit modifying a `DefaultAttribute` instance concurrently by multiple threads so creating a lock in this method is not going to work.
Instead, we need to add a lock to `DefaultAttribute` or make `DefaultAttribute` to extend `ReentrantLock` not to create an additional instance.
```java
head.lock();
...
head.unlock();

static final class DefaultAttribute<T> extends ReentrantLock implements Entry<AttributeKey<T>, T> {...}
```

Reply:
As mentioned, added a lock.